services/galley/src/Galley/Data.hs

```diff
@@ -805,3 +836,26 @@ lookupClients users =
 
 eraseClients :: MonadClient m => UserId -> m ()
 eraseClients user = retry x5 (write Cql.rmClients (params Quorum (Identity user)))
+
+-- Internal utilities
+newTeamMember' :: (MonadThrow m, MonadClient m) => (UserId, Permissions, Maybe UserId, Maybe UTCTimeMillis, Maybe UserLegalHoldStatus) -> m TeamMember
+newTeamMember' (uid, perms, minvu, minvt, mlhStatus) = newTeamMemberRaw uid perms minvu minvt (fromMaybe UserLegalHoldDisabled mlhStatus)
+
+-- | Invoke the given continuation 'k' with a list of TeamMemberRows IDs
+-- which are looked up based on:
+type TeamMemberRow = (UserId, Permissions, Maybe UserId, Maybe UTCTimeMillis, Maybe UserLegalHoldStatus)
+
+withTeamMembers ::
+  TeamId ->
+  ([TeamMember] -> Galley ()) ->
+  Galley ()
+withTeamMembers tid k = do
```

Comment:
Looking at the call site of this function, I didn't know that it would call the continuation multiple times, on chunks of team members. Maybe this could be clearer from the name? Something like `withTeamMemberChunks`?

Reply:
> Looking at the call site of this function, I didn't know that it would call the continuation multiple times, on chunks of team members. Maybe this could be clearer from the name? Something like `withTeamMemberChunks`?

👌